### ISSUE TITLE
Arm64 macOS support

### ISSUE THREAD

**Issue by eed3si9n** [swoval/swoval#141]
This was originally reported by @catap in https://github.com/sbt/sbt/issues/6162#issuecomment-732510615

## steps

sbt/sbt@13b09bcd as root published locally as `1.5.0-SNAPSHOT`.

## problem

```
catap@Kirills-mini-m1 sbt % java -Dsbt.version=1.5.0-SNAPSHOT -jar /opt/local/share/sbt/sbt-launch.jar
[info] welcome to sbt 1.5.0-SNAPSHOT (Azul Systems, Inc. Java 16-ea)
[info] loading settings for project sbt-build from plugins.sbt ...
[info] loading project definition from /Users/catap/src/sbt/project
[info] compiling 11 Scala sources to /Users/catap/src/sbt/project/target/scala-2.12/sbt-1.0/classes ...
[info] Non-compiled module 'compiler-bridge_2.12' for Scala 2.12.12. Compiling...
[info]   Compilation completed in 4.319s.
[warn] /Users/catap/src/sbt/project/SbtLauncherPlugin.scala:50:10: method zip in object IO is deprecated (since 1.3.2): Please specify whether to use a static timestamp
[warn]       IO.zip(dir.allPaths.get().flatMap(rebase), target)
[warn]          ^
[warn] /Users/catap/src/sbt/project/Scripted.scala:111:37: method toLoader in object ClasspathUtilities is deprecated (since 1.4.0): internal
[warn]     val loader = ClasspathUtilities.toLoader(classpath, noJLine)
[warn]                                     ^
[warn] /Users/catap/src/sbt/project/Scripted.scala:111:18: object ClasspathUtilities in package classpath is deprecated (since 1.4.0): internal
[warn]     val loader = ClasspathUtilities.toLoader(classpath, noJLine)
[warn]                  ^
[warn] three warnings found
[info] loading settings for project sbtRoot from build.sbt ...
[info] resolving key references (35915 settings) ...
[info]            __    __ 
[info]      _____/ /_  / /_
[info]     / ___/ __ \/ __/
[info]    (__  ) /_/ / /_  
[info]   /____/_.___/\__/ 
[info] Welcome to the build for sbt.
[info] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[info]   Java version is 16. We recommend java 8.
[info] !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Couldn't load native library java.lang.UnsatisfiedLinkError: /Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: dlopen(/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib, 1): no suitable image found.  Did find:
	/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: mach-o, but wrong architecture
	/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: stat() failed with errno=17
java.lang.ExceptionInInitializerError
	at com.swoval.files.apple.FileEventMonitorImpl.<clinit>(FileEventMonitors.java:130)
	at com.swoval.files.apple.FileEventMonitors.get(FileEventMonitors.java:47)
	at com.swoval.files.ApplePathWatcher.<init>(ApplePathWatcher.java:258)
	at com.swoval.files.ApplePathWatcher.<init>(ApplePathWatcher.java:194)
	at com.swoval.files.ApplePathWatchers.get(ApplePathWatcher.java:331)
	at com.swoval.files.PathWatchers.get(PathWatchers.java:84)
	at com.swoval.files.FileTreeRepositories.get(FileTreeRepositories.java:64)
	at com.swoval.files.FileTreeRepositories.get(FileTreeRepositories.java:32)
	at sbt.internal.nio.FileTreeRepositoryImpl.<init>(FileTreeRepositoryImpl.scala:46)
	at sbt.internal.nio.FileTreeRepository$.default(FileTreeRepository.scala:40)
	at sbt.BuiltinCommands$.$anonfun$setupGlobalFileTreeRepository$1(Main.scala:954)
	at sbt.BuiltinCommands$.doLoadProject(Main.scala:945)
	at sbt.BuiltinCommands$.$anonfun$loadProjectImpl$2(Main.scala:885)
	at sbt.Command$.$anonfun$applyEffect$4(Command.scala:150)
	at sbt.Command$.$anonfun$applyEffect$2(Command.scala:145)
	at sbt.Command$.process(Command.scala:189)
	at sbt.MainLoop$.$anonfun$processCommand$5(MainLoop.scala:242)
	at scala.Option.getOrElse(Option.scala:189)
	at sbt.MainLoop$.process$1(MainLoop.scala:242)
	at sbt.MainLoop$.processCommand(MainLoop.scala:268)
	at sbt.MainLoop$.$anonfun$next$5(MainLoop.scala:162)
	at sbt.State$StateOpsImpl$.runCmd$1(State.scala:289)
	at sbt.State$StateOpsImpl$.process$extension(State.scala:325)
	at sbt.MainLoop$.$anonfun$next$4(MainLoop.scala:162)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
	at sbt.MainLoop$.next(MainLoop.scala:162)
	at sbt.MainLoop$.run(MainLoop.scala:143)
	at sbt.MainLoop$.$anonfun$runWithNewLog$1(MainLoop.scala:118)
	at sbt.io.Using.apply(Using.scala:27)
	at sbt.MainLoop$.runWithNewLog(MainLoop.scala:111)
	at sbt.MainLoop$.runAndClearLast(MainLoop.scala:65)
	at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:50)
	at sbt.MainLoop$.runLogged(MainLoop.scala:41)
	at sbt.StandardMain$.runManaged(Main.scala:191)
	at sbt.xMain$.$anonfun$run$8(Main.scala:102)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withIn(Console.scala:230)
	at sbt.internal.util.Terminal$.withIn(Terminal.scala:557)
	at sbt.internal.util.Terminal$.$anonfun$withStreams$1(Terminal.scala:347)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withOut(Console.scala:167)
	at sbt.internal.util.Terminal$.$anonfun$withOut$2(Terminal.scala:547)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at scala.Console$.withErr(Console.scala:196)
	at sbt.internal.util.Terminal$.withOut(Terminal.scala:547)
	at sbt.internal.util.Terminal$.withStreams(Terminal.scala:347)
	at sbt.xMain$.run(Main.scala:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:83)
	at sbt.xMain.run(Main.scala:46)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
	at xsbt.boot.Launch$.run(Launch.scala:109)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
	at xsbt.boot.Launch$.launch(Launch.scala:117)
	at xsbt.boot.Launch$.apply(Launch.scala:18)
	at xsbt.boot.Boot$.runImpl(Boot.scala:41)
	at xsbt.boot.Boot$.main(Boot.scala:17)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.UnsatisfiedLinkError: /Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: dlopen(/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib, 1): no suitable image found.  Did find:
	/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: mach-o, but wrong architecture
	/Users/catap/.sbt/1.0/swoval-jni/8091807251952918964-libswoval-files0.dylib: stat() failed with errno=17
	at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:383)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:227)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:169)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2383)
	at java.base/java.lang.Runtime.load0(Runtime.java:746)
	at java.base/java.lang.System.load(System.java:1857)
	at com.swoval.runtime.NativeLoader.loadPackaged(NativeLoader.java:140)
	at com.swoval.runtime.NativeLoader.loadPackaged(NativeLoader.java:174)
	at com.swoval.files.apple.FileEventMonitorImpl.<clinit>(FileEventMonitors.java:127)
	... 61 more
[error] java.lang.ExceptionInInitializerError
[error] Use 'last' for the full log.
```

**Comment by eatkins** [comment]
It looks like it's possible to build a universal library for x86_64 and arm64: https://developer.apple.com/documentation/xcode/building_a_universal_macos_binary. I just have to get all of the xcode stuff set up.

**Comment by catap** [comment]
@eatkins I do have everything in the place to make a fat binary. Let me help :)

**Comment by catap** [comment]
@eatkins let me know when I can make the next attempt ;)

**Comment by eatkins** [comment]
Should be fixed by #142.
